tst/EnergyPlus/unit/PackagedTerminalHeatPump.unit.cc

```diff
@@ -4359,4 +4360,328 @@ TEST_F(EnergyPlusFixture, PTAC_ZoneEquipment_NodeInputTest)
+    // set thermostat type
+    state->dataHeatBalFanSys->TempControlType.allocate(1);
+    state->dataHeatBalFanSys->TempControlTypeRpt.allocate(1);
+    state->dataHeatBalFanSys->TempControlType(1) = DataHVACGlobals::ThermostatType::DualSetPointWithDeadBand;
+    // set the uni sys is always available
+    state->dataScheduleMgr->Schedule(state->dataUnitarySystems->unitarySys[0].m_SysAvailSchedPtr).CurrentValue = 1.0;
+    state->dataScheduleMgr->Schedule(state->dataUnitarySystems->unitarySys[0].m_FanAvailSchedPtr).CurrentValue = 1.0;
+
+    bool HeatActive = true;
+    bool CoolActive = false;
+    bool FirstHVACIteration = false;
+    int CompIndex = 1;
+    int AirLoopNum = 0;
+    int ControlZoneNum = 1;
+    int constexpr ZoneOAUnitNum = 0;
+    Real64 sensOut = 0.0;
+    Real64 latOut = 0.0;
+    Real64 constexpr OAUCoilOutTemp = 0.0;
+    // set local variables for convenience
+    auto &supplyFan = state->dataFans->Fan(1);
+    auto &dxClgCoilMain = state->dataDXCoils->DXCoil(1);
+    auto &dxHtgCoilMain = state->dataDXCoils->DXCoil(2);
+    auto &elecHtgCoilSupp = state->dataHeatingCoils->HeatingCoil(1);
+
+    state->dataGlobal->BeginEnvrnFlag = true;
+    mySys->simulate(*state,
+                    thisSys.Name,
+                    FirstHVACIteration,
+                    AirLoopNum,
+                    CompIndex,
+                    HeatActive,
+                    CoolActive,
+                    ZoneOAUnitNum,
+                    OAUCoilOutTemp,
+                    zoneEquipment,
+                    sensOut,
+                    latOut);
+
+    EXPECT_TRUE(state->dataUnitarySystems->HeatingLoad);
+    EXPECT_FALSE(state->dataUnitarySystems->CoolingLoad);
+
+    Real64 result_pthp_ElectricityRate = supplyFan.FanPower + state->dataHVACGlobal->DXElecCoolingPower + state->dataHVACGlobal->DXElecHeatingPower +
+                                         state->dataHVACGlobal->DefrostElecPower + state->dataHVACGlobal->ElecHeatingCoilPower +
+                                         state->dataHVACGlobal->SuppHeatingCoilPower + thisSys.m_TotalAuxElecPower;
+    // test results
+    EXPECT_NEAR(30636.88, thisSys.m_ElecPower, 0.01);
+    EXPECT_NEAR(30636.88, result_pthp_ElectricityRate, 0.01);
+    EXPECT_NEAR(857.14, supplyFan.FanPower, 0.01);
+    EXPECT_NEAR(0.0, dxClgCoilMain.ElecCoolingPower, 0.01);
+    EXPECT_NEAR(0.0, state->dataHVACGlobal->DXElecCoolingPower, 0.01);
+    EXPECT_NEAR(4327.88, dxHtgCoilMain.ElecHeatingPower, 0.01);
+    EXPECT_NEAR(4327.88, state->dataHVACGlobal->DXElecHeatingPower, 0.01);
+    EXPECT_NEAR(0.0, state->dataHVACGlobal->ElecHeatingCoilPower, 0.01);
+    EXPECT_NEAR(333.33, dxHtgCoilMain.DefrostPower, 0.01);
+    EXPECT_NEAR(333.33, state->dataHVACGlobal->DefrostElecPower, 0.01);
+    EXPECT_NEAR(25118.52, elecHtgCoilSupp.ElecUseRate, 0.01);
+    EXPECT_NEAR(25118.52, state->dataHVACGlobal->SuppHeatingCoilPower, 0.01);
+    EXPECT_NEAR(0.0, thisSys.m_TotalAuxElecPower, 0.01);
+}
+
```

Comment:
Unit test was added for issue #9839 illustrating that the parent total `electricity rate` matches sum of the components including `defrost electric power`.